js/reference.js

```diff
@@ -747,21 +747,36 @@ var ERMrest = (function(module) {
          * @type {string}
          */
         get displayname() {
-            // TODO: what is row name if no annotation?
-            // if (this._displayname === undefined) {
-            //     var tableDisplayAnnotation = "tag:isrd.isi.edu,2016:table-display";
-            //     if(this._ref.table.annotations.contains(tableDisplayAnnotation)) {
-            //         var annotation = this._ref.table.annotations.get(tableDisplayAnnotation);
-            //         if (annotation.row_name) {
-            //             var pattern = annotation.row_name;
-            //             // TODO: parse the pattern to get all `{***}` tokens
-            //             //    -  implement pattern expansion funcion?
-            //         }
-            //     }
-            // }
-            //
-            // return this._displayname;
+            if (!this._displayname) {
+                var table = this._ref._table;
+                // if table has display row_name annotation
+                if (table.annotations.contains("tag:isrd.isi.edu,2016:table-display") &&
```

Comment:
@shinyichen, Aref introduced an enumeration for annotation names and we decided that we will add any in-use annoation to that list and then use the enum like `module._annotations.TABLE_DISPLAY` for example. You can see that enum here:

https://github.com/informatics-isi-edu/ermrestjs/blob/280944127798b7ddd07762838e33d45f1e58aa6c/js/utilities.js#L329